src/main/java/com/sap/oss/phosphor/fosstars/model/value/Vulnerability.java

```diff
@@ -104,9 +105,9 @@
   private final Date published;
 
   /**
-   * List of vulnerable version ranges.
+   * List of vulnerable artifacts.
    */
-  private final List<VersionRange> vulnerableVersions;
+  private final List<VulnerableArtifact> vulnerableArtifacts;
```

Comment:
This new class refers to NVD. But the `Vulnerability` class doesn't represent data from NVD. In fact, it describes a vulnerability, not an entry from NVD. NVD is not the only available database of vulnerabilities. Let's please not make `Vulnerability` specific to NVD.

I am not sure if `Vulnerability` needs to hold a list of artifacts (please see https://github.com/SAP/fosstars-rating-core/issues/487#issuecomment-809531836). This class already holds quite a lot of info. Plus, I am not sure if wee need `VulnerableArtifact` at all. You introduce `Artifact` and its implementations in another pull request. How about adding `AbstractArtifact` that implements `Artifact` and holds a list of vulnerabilities?